web/src/lib/components/shared-components/version-announcement-box.svelte

```diff
@@ -15,22 +16,30 @@
     return `v${major}.${minor}.${patch}`;
   }
 
-  function onAcknowledge() {
+  const onAcknowledge = () => {
     // Store server version to prevent the notification
     // from showing again.
     localStorage.setItem('appVersion', githubVersion);
     showModal = false;
-  }
+  };
 
   onMount(async () => {
     try {
-      githubVersion = await getGithubVersion();
+      const { data } = await api.userApi.getLatestImmichVersionAvailable();
+      if (data.availableVersion) {
+        githubVersion = semverToName({
+          major: data.availableVersion.major,
+          minor: data.availableVersion.minor,
+          patch: data.availableVersion.patch,
+        });
+      }
```

Comment:
IMO the server should be the one that determines if a new version is available. I don't think it makes sense for each client to have to manually do the computation. We could just return something like:

```typescript
{
  isAvailable: boolean;
  releaseType: 'patch' | 'minor' | 'major' | 'none',
  currentVersion: IServerVersion;
  releaseVersion: IServerVersion;
  
}
```

Reply:
what is releaseType for?